File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/variantbeaufort/VariantBeaufortSuite.kt
package cryptobench.ciphers.variantbeaufort

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class VariantBeaufortSuite : CipherSuite {
    override fun name(): String = "variantbeaufort"

    override fun run(): SuiteResult {
        val cipher: Cipher = VariantBeaufortCipher(VariantBeaufortKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in VariantBeaufortVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}

File: fixtures/scale/dart/large/lib/ciphers/variantbeaufort/variantbeaufort_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'variantbeaufort_cipher.dart';
import 'variantbeaufort_key.dart';
import 'variantbeaufort_vectors.dart';

class VariantBeaufortSuite implements CipherSuite {
  @override
  String name() => 'variantbeaufort';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = VariantBeaufortCipher(VariantBeaufortKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in variantbeaufortSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
